Api for removing multiple users from multiple courses

Imran92 opened the pull request:
Fixes part of https://github.com/Automattic/sensei/issues/4959

### Changes proposed in this Pull Request
Add REST API endpoint to add users to courses

### Testing instructions
- Create a few users and a few courses

- Enroll those users to those courses, either from the UI or by using the rest API here https://github.com/Automattic/sensei/pull/4968

- Get WP REST nonce (make sure you're logged in): http://devwp.local/wp-admin/admin-ajax.php?action=rest-nonce (replace devwp.local with your domain)

- Copy Cookie name and value
- Execute in terminal (replace WPNONCE, COOKIE-NAME & COOKIE-VALUE, USER_ID, COURSE_ID with actual values):
```
curl -X "DELETE" "http://devwp.local/?rest_route=%2Fsensei-internal%2Fv1%2Fcourse-students%2Fbatch&_wpnonce=WPNONCE" \
     -H 'Cookie: COOKIE-NAME=COOKIE-VALUE' \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{
  "student_ids": USER_ID,
  "course_ids": COURSE_ID
}'
```
Make sure you get status code 200 and your users are removed from the courses.

If you use vscode's thunder client, you can import the following collection for testing the API. Change the URL and the variables as per your system config
[thunder-collection_Sensei.json.zip](https://github.com/Automattic/sensei/files/8428461/thunder-collection_Sensei.json.zip)

Commit:
Api for removing multiple users from multiple courses

Commit:
Merge branch 'add/rest-api-add-to-course' into add/rest-api-remove-from-course

Commit:
Shift api to batch, add request schema, update test

Commit:
Merge branch 'add/rest-api-add-to-course' into add/rest-api-remove-from-course

Commit:
Standardize function names

merkushin commented on includes/rest-api/class-sensei-rest-api-course-users-controller.php:
```diff
@@ -81,20 +87,47 @@ public function batch_create_items( WP_REST_Request $request ) {
-	 * @return bool
+	 * @return bool|WP_Error
 	 */
 	public function batch_create_items_permissions_check( WP_REST_Request $request ): bool {
```
I think we need a better name for this method :)
Maybe `batch_permissions_check`?

merkushin commented on tests/unit-tests/rest-api/test-class-sensei-rest-api-course-users-controller.php:
```diff
@@ -117,4 +117,45 @@ public function testAddUsersToCourses_RequestGiven_EnrolsUserForCourse() {
 		$this->assertTrue( $enrolment->is_enrolled( $user_id ) );
 	}
+
+	public function testRemoveUsersFromCoursesApi_AfterApiExecution_StudentsAreActuallyRemoved() {
```
[As Gabriel suggested in my PR](https://github.com/Automattic/sensei/pull/4968#discussion_r843274952), it might be a good idea to add tests for sad paths. WDYT?

merkushin commented:
Hmm, DELETE method doesn't work for me for some reason right now (I get 405 Not Allowed). I suppose that's because of Local configuration. I'll check it one more time a bit later.

Commit:
Merge and add tests for sad paths

Commit:
Merge and add tests for sad path updated

Imran92 commented:
Hi @merkushin , Thanks for your suggestions. I've updated the PR accordingly, can you kindly take another look? I've also added a zip in the description that you can use with vscode's Thunder Client for testing the API

merkushin commented:
The problem with executing custom requests (I mean custom methods, like DELETE) was somewhere in the Nginx configuration. I failed finding the exact problem, but solved it by switching from nginx to Apache.

merkushin reviewed (approved):
Looks good and works well 👍
